cfpb/capital-framework
# Big atomic update to cf-icons

Default usage of icons required an empty elem because the font family was set on the elem level. This change restricts the font-family to the pseudo-elem allowing icons to be used with standard text. This change will eliminate the duplicated icon classes in cf-buttons and cf-typography. Icon only elems can still be used by adding `cf-icon__before` the existing class list.

## Changes

- Updated the `cf-icon` class to be a mixin only, so that it's not duplicated when `cf-icon-prefix` is set as `cf-icon`.
- Updated the pseudo elem to only include an icon when a modifier class is used, this is important to mix icons into elems with normal text like buttons, links, etc.
- Removed IE7 support because the hack stopped working.

## Testing

- check out this branch and run `npm run cf-link`
- check out Sandbox branch `example/cf-icons` and run `npm run cf-link`
- run `npm start`
- check `http://localhost:3000/components/cf-icons/`

## Review

- @Scotchester 
- @KimberlyMunoz 

## Screenshots

Everything's the same

## Notes

- Adding the `__before` modifier is a small but necessary hurdle to making this possible. The benefit of removing 20kb of duplicated code outweighs that extra work.

## Todos

- Remove the duplicated icon code through the rest of the project.

## Checklist
